### DISCUSSION TITLE
Possible to style individual app icons in launcher?

### DISCUSSION THREAD

**Discussion by zoenetic** [JakeStanger/ironbar#763] [Q&A]
I was wondering if it's possible to add CSS styles to specific icons within the launcher widget on a per-app basis (based on app id, name, or something like that). I had a poke around with ironbar inspect, but couldn't see any relevant classes. 

I was looking for this as a way to replace the stock icons for given apps with those in nerdfonts, or resources like that, via CSS. So any alternative ideas for achieving that would be appreciated.

**Comment by JakeStanger** [comment]
Currently this isn't possible I'm afraid. It should be easy enough to add the app id as the widget name to make it targetable. Adding a mechanism for replacing the images is a bit harder and would need more thought, but I'm open to ideas. Feel free to open a feature request issue!

**Reply by zoenetic** [reply]
Thanks, I'll log a feature request for making the app id the widget name.

As for replacing the icon, I was thinking about the content property in CSS, but it looks like GTK 3's version of CSS doesn't implement that. Alternatively perhaps there could be an option for the launcher which sets the format of each app's name, and which accepts the app's ID as an input?

So you could have something like:

```
- type: launcher
  show_names: true
  show_icons: false
  name_format: {{script which outputs nerdfont for $app_id here}}
```
